src/PhpSpec/ObjectBehavior.php

```diff
@@ -31,9 +31,90 @@
  * @method void beConstructedWith(...$arguments)
  * @method void beConstructedThrough($factoryMethod, array $constructorArguments = array())
  * @method void beAnInstanceOf($class)
+ *
  * @method void shouldHaveType($type)
+ * @method void shouldNotHaveType($type)
+ * @method void shouldReturnAnInstanceOf($type)
```

Comment:
@stof I am aware of that and those magic methods on `Subject` are intended to help building IDE support via a plugin. As I told before already, I already have a WIP plugin for PhpStorm which collects the SUT methods and makes them available via `$this`. Those methods then are typed to return `Subject`. The current problem is that we are stuck at this place because we do not have the available matcher methods as those are not present in the parse tree (missing `@method` annotations). And there currently is no way to add stub elements dynamically without having them somewhere in the code. There it is mandatory to have those. Also I am not sure how it would be possible to "support matchers properly rather than relying on `@mathod`" as we do not the matcher method names, arguments and return types. How would you infer that dynamically?